src/main/cues/cue-emitter.js

```diff
@@ -0,0 +1,222 @@
+/**
+ * @file
+ * @module peaks/cues/cue-emitter
+ */
+define([
+  'peaks/cues/cuemark'
+], function(CueMark) {
+  'use strict';
+
+  var isHeadless = /HeadlessChrome/.test(navigator.userAgent);
+
+  function windowIsVisible() {
+    if (isHeadless || navigator.webdriver) {
+      return false;
+    }
+    return (typeof document === 'object') &&
+      ('visibilityState' in document) &&
+      (document.visibilityState === 'visible');
+  }
+
+  function getRAF(top) {
+    return top ? (
+      top.requestAnimationFrame ||
+      top.webkitRequestAnimationFrame ||
+      top.mozRequestAnimationFrame ||
+      top.msRequestAnimationFrame ||
+      function() {}
+    ) : function() {};
+  }
+  var rAF = getRAF(window);
+
+  /**
+   * this adapter navigates the peaks instance to get a corresponding point or segment
+   * @param {Peaks} peaks
+   * @param {CueMark} mark
+   * @return {Point|Segment}
+   * @throws {Error}
+   */
+  function getPointOrSegment(peaks, mark) {
+     switch (mark.type) {
+      case CueMark.POINT:
+        return peaks.points.getPoint(mark.id);
+      case CueMark.SEGMENT_START:
+      case CueMark.SEGMENT_END:
+        return peaks.segments.getSegment(mark.id);
+       default:
+         throw new Error('getPointOrSegment: id not found?');
+    }
+  }
+
+  /**
+   * @class
+   * @property {Peaks} peaks
+   * @property {Array.<CueMark>} marks
+   * @param {Peaks} peaks
+   * @constructor
+   */
+  function CueEmitter(peaks) {
+    this.marks = Array();
```

Comment:
We're moving to a naming convention where private fields have a leading underscore. If you prefer, I can make this change?